pkg/apis/options/providers.go

```diff
@@ -29,7 +31,7 @@ type Provider struct {
 	// KeycloakConfig holds all configurations for Keycloak provider.
 	KeycloakConfig KeycloakOptions `json:"keycloakConfig,omitempty"`
 	// AzureConfig holds all configurations for Azure provider.
-	AzureConfig AzureOptions `json:"azureConfig,omitempty"`
+	AzureConfig *AzureOptions `json:"azureConfig,omitempty"`
```

Comment:
As we have removed the Azure defaults, we can move this back to not being a pointer